From 83d54be0632f844a834385e4d2517aabdfb9cb6c Mon Sep 17 00:00:00 2001
From: Daniela Plascencia <daniela.plascencia@linux.intel.com>
Date: Mon, 20 Aug 2018 17:23:22 -0500
Subject: [PATCH] Set kata-runtime as default runtime for cri-o.

As it was done for Docker, the runtime configuration for cri-o will be done
automatically when Kata Containers is installed on a Clear Linux system.

Signed-off-by: Daniela Plascencia <daniela.plascencia@linux.intel.com>
---
 crio-clearlinux.conf     |  3 +++
 crio-set-runtime.service |  8 ++++++++
 set-crio-default-runtime | 14 ++++++++++++++
 3 files changed, 25 insertions(+)
 create mode 100644 crio-clearlinux.conf
 create mode 100644 crio-set-runtime.service
 create mode 100644 set-crio-default-runtime

diff --git a/crio-clearlinux.conf b/crio-clearlinux.conf
new file mode 100644
index 0000000..24542b9
--- /dev/null
+++ b/crio-clearlinux.conf
@@ -0,0 +1,3 @@
+[Unit]
+Wants=crio-set-runtime.service
+After=crio-set-runtime.service
diff --git a/crio-set-runtime.service b/crio-set-runtime.service
new file mode 100644
index 0000000..b0c7709
--- /dev/null
+++ b/crio-set-runtime.service
@@ -0,0 +1,8 @@
+[Unit]
+Description= Set Default Cri-o Runtime
+ConditionPathExists=!/etc/crio/crio.conf
+ConditionPathExists=/dev/kvm
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/set-crio-default-runtime
diff --git a/set-crio-default-runtime b/set-crio-default-runtime
new file mode 100644
index 0000000..e8ea8e8
--- /dev/null
+++ b/set-crio-default-runtime
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+mkdir -p /etc/crio/
+mkdir -p /opt/cni/bin/
+
+cp /usr/share/defaults/crio/crio.conf /etc/crio/crio.conf
+
+sed -i 's|^\(runtime_untrusted_workload = \).*|\1"/usr/bin/kata-runtime"|' /etc/crio/crio.conf
+
+sed -i 's|\(\[crio\.runtime\]\)|\1\nmanage_network_ns_lifecycle = true|' /etc/crio/crio.conf
+
+sed -i 's|/usr/libexec/cni/|/opt/cni/bin/|' /etc/crio/crio.conf
+
+systemctl daemon-reload
-- 
2.19.1

